src/ncp/ncp_base.cpp

```diff
@@ -3494,13 +3544,53 @@ ThreadError NcpBase::SetPropertyHandler_MAC_SCAN_STATE(uint8_t header, spinel_pr
             break;
 
         case SPINEL_SCAN_STATE_ENERGY:
-            errorCode = otEnergyScan(
-                mInstance,
-                mChannelMask,
-                mScanPeriod,
-                &HandleEnergyScanResult_Jump,
-                this
-            );
+#if OPENTHREAD_ENABLE_RAW_LINK_API
+            if (otLinkRawIsEnabled(mInstance))
+            {
+                if (mCurScanChannel == NCP_INVALID_SCAN_CHANNEL)
+                {
+                    uint8_t scanChannel;
+                    uint16_t scanDuration;
+
+                    parsedLength = spinel_datatype_unpack(
```

Comment:
I thought about doing that, but I ended up specifically going that route for a few reasons:

1. It seemed inefficient to separate out a single function call into multiple individual commands just to pass the arguments. The Host is then required to make several async function calls (to send commands) and wait for their results in series.
2. It didn't seem like a good design to reuse `mScanChannelMask` here because that was specifically designed as a bit field for a set of channels. I also didn't think we should add a new variable, as that would require yet another command to get/set it. Finally, there is no reason for the variable to be persisted in RAM, as it is completely temporary.
3. Besides scans, I couldn't find another OpenThread API that split it's calling logic between multiple function calls. For instance, all the _STREAM commands take all parameters inline.